### PR TITLE
SSO extension extra params

### PR DESCRIPTION
**Pull request by antrix1989**

### CODE REVIEW

File: IdentityCore/src/broker_operation/request/token_request/MSIDBrokerOperationTokenRequest.m [hunk 1]
```diff
@@ -63,6 +75,35 @@ - (instancetype)initWithJSONDictionary:(NSDictionary *)json error:(NSError **)er
...
+            _mamResources = (NSDictionary *)[[MSIDJsonSerializer new] fromJsonString:mamResourcesStr ofType:NSDictionary.class context:nil error:nil];
+        }
+        
+        NSString *clientCapabilitiesStr = [json msidStringObjectForKey:MSID_BROKER_CLIENT_CAPABILITIES_KEY];
+        _clientCapabilities = [clientCapabilitiesStr componentsSeparatedByString:@","];
+        
+        NSString *claimsStr= [json msidStringObjectForKey:MSID_BROKER_CLAIMS_KEY];
+        if (claimsStr)
+        {
+            _claimsRequest = (MSIDClaimsRequest *)[[MSIDJsonSerializer new] fromJsonString:claimsStr ofType:MSIDClaimsRequest.class context:nil error:nil];
```
Comment (oldalton) [inline IdentityCore/src/broker_operation/request/token_request/MSIDBrokerOperationTokenRequest.m]: nit: can we log an error for debug purposes?

File: IdentityCore/src/broker_operation/request/token_request/MSIDBrokerOperationTokenRequest.m [hunk 2]
```diff
@@ -77,7 +118,14 @@ - (NSDictionary *)jsonDictionary
     if (!configurationJson) return nil;
     [json addEntriesFromDictionary:configurationJson];
     json[MSID_PROVIDER_TYPE_JSON_KEY] = MSIDProviderTypeToString(self.providerType);
-    
+    json[MSID_BROKER_EXTRA_OIDC_SCOPES_KEY] = self.oidcScope;
+    json[MSID_BROKER_EXTRA_QUERY_PARAM_KEY] = self.extraQueryParameters;
+    json[MSID_BROKER_INSTANCE_AWARE_KEY] = [@(self.instanceAware) stringValue];
+    json[MSID_BROKER_INTUNE_ENROLLMENT_IDS_KEY] = [self.enrollmentIds msidJSONSerializeWithContext:nil];
+    json[MSID_BROKER_INTUNE_MAM_RESOURCE_KEY] = [self.mamResources msidJSONSerializeWithContext:nil];
+    json[MSID_BROKER_CLIENT_CAPABILITIES_KEY] = [self.clientCapabilities componentsJoinedByString:@","];
```
Comment (oldalton) [inline IdentityCore/src/broker_operation/request/token_request/MSIDBrokerOperationTokenRequest.m]: would it be safer to url encode this? This is technically any random string from the app side.

File: IdentityCore/src/requests/broker/MSIDSSOExtensionInteractiveTokenRequest.m [hunk 3]
```diff
@@ -104,9 +110,27 @@ - (void)executeRequestWithCompletion:(MSIDInteractiveRequestCompletionBlock)comp
          }
         
         NSError *localError;
+        NSDictionary *enrollmentIds = [self.enrollmentIdsCache enrollmentIdsJsonDictionaryWithContext:self.requestParameters
+                                                                                                error:&localError];
+        if (localError)
```
Comment (oldalton) [inline IdentityCore/src/requests/broker/MSIDSSOExtensionInteractiveTokenRequest.m]: Should we consider this error as non critical?

File: IdentityCore/src/requests/broker/MSIDSSOExtensionInteractiveTokenRequest.m [hunk 4]
```diff
@@ -104,9 +110,27 @@ - (void)executeRequestWithCompletion:(MSIDInteractiveRequestCompletionBlock)comp
...
+        {
+            completionBlock(nil, localError, nil);
+            return;
+        }
+        
+        NSDictionary *mamResources = [self.mamResourcesCache resourcesJsonDictionaryWithContext:self.requestParameters
+                                                                                          error:&localError];
+        if (localError)
```
Comment (oldalton) [inline IdentityCore/src/requests/broker/MSIDSSOExtensionInteractiveTokenRequest.m]: same here

File: IdentityCore/src/broker_operation/request/token_request/MSIDBrokerOperationTokenRequest.m (hunk 1 above)
Comment (antrix1989) [reply IdentityCore/src/broker_operation/request/token_request/MSIDBrokerOperationTokenRequest.m]: We log in json serializer.

File: IdentityCore/src/broker_operation/request/token_request/MSIDBrokerOperationTokenRequest.m (hunk 2 above)
Comment (antrix1989) [reply IdentityCore/src/broker_operation/request/token_request/MSIDBrokerOperationTokenRequest.m]: URLQueryItem will url encode all params.

File: IdentityCore/src/requests/broker/MSIDSSOExtensionInteractiveTokenRequest.m (hunk 3 above)
Comment (antrix1989) [reply IdentityCore/src/requests/broker/MSIDSSOExtensionInteractiveTokenRequest.m]: done

File: IdentityCore/src/requests/broker/MSIDSSOExtensionInteractiveTokenRequest.m (hunk 4 above)
Comment (antrix1989) [reply IdentityCore/src/requests/broker/MSIDSSOExtensionInteractiveTokenRequest.m]: done

**Pull request merged**